### DISCUSSION TITLE
Algorithm to detect trending repositories

### DISCUSSION THREAD

**Discussion by kkoomen** [community/community#3083] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi,

I want to know suggest and discuss possible optimizations for the Trending algorithm that resides on the https://github.com/trending page, because I don't know how it's being calculated right now, but to me there is definitely something off.

I have a vim plugin called [vim-doge](https://github.com/kkoomen/vim-doge) where I worked on many months before making it public. After making it public and announcing it Reddit's [/r/vim](https://reddit.com/r/vim) I did got like 100 stars just over a night of sleep and because of that, the algorithm detected that perhaps something "special" was happening to my repository and GitHub did put it in [the trending page of vim](https://github.com/trending/vim).

Nowadays, a year or 2 later, I still get some stars here on a daily basis, yet it just doesn't end up in the trending page, but in the trending page there are other repositories that haven't received stars in days and they are in the trending page for a few days!

How's this possible?!

These repositories are quite popular already and thus so many people have starred them that they rarely get any stars. Even my repository does get more stars than them daily and yet, those repositories always seem to be in the trending pages.

So, the trending page did got a redesign last year, but I am not sure when's the last time the algorithm had a redesign.

So, before going into scenarios, what does *trending* mean? Here's a definition from dictionary.com:

> trending / ˈtrɛn dɪŋ /
> widely mentioned or discussed on the internet

So, whenever I am looking at trending pages, I should see things that are popular on GitHub. Things that are being used a lot, receive lots of stars on a daily basis and perhaps are very active in maintaining the repository/product itself. Now, the trending page would also be a great place to find small, but awesome projects that did receive quite stars in a short amount of time, but just not enough to reach trending, because obviously the big projects will still receive more stars in a shorter amount of time than the smaller ones, even though it's still possible that the smaller projects can be above this.

GitHub has lots of info on a repository so there can be many factors that can decide when something is popular. Maybe the amount of forks, stars, issues, pull requests or contributions in the past 24h. I don't know. I'm just throwing out any kind of idea that I can think of.

Personally, I think that it's important that the trending page is dynamic, but unfortunately, when I check it daily, most of the time it's always the same big names that are showing up. It's important that small projects can get attention and are able to be discovered by people. Some projects are being developed in order to solve problems in our world and help mankind, but they're being pushed away because "The Big 5" are on trending most of time or other well-funded projects, which makes sense as they just gets lots of stars in a single day. Some of them get thousands of stars in a single day.

One of the most important factors that the trending page (for specific languages) has is not being able to provide accurate results. Why does my repository in `/trending/vim` that received 20+ stars in a day not end up in trending, while there are projects in `/trending/vim` for a week where they get less than 5 stars in a week?!

I think, in order to discover some of the less-popular repositories, the minimum but probably one of the most valueable factor should be the amount of stars in the shortest amount of time.

The second factor would probably be the time the repository got created (to know whether this is a new repository or an older one that has existed for many years). If this is a repository of a week old and it already received 100 stars in 2 days, then that's quite impressive and there's a good chance this can be something that lots of people will enjoy.

The third factor can be a calculation of how active the repository has been in the last day, week or month. This can, for example, be based on the commits / tags + amount of merged PRs + amount of created PRs + comments/replies by the author. The can result into a score and the higher the score, the higher the probability it will go trending if you combine with with factor 1 (the amount of stars in a short time) and factor 2 (when it got created).

These are just ideas and I hope I can help other people who read this to find solutions based on my suggestions and help GitHub to let the world discover each others work.

**Comment by evi-liu** [comment, marked as answer]
This forum is for feedback specific to GitHub Discussions, but I've passed along your feedback/question to our team that owns `Trending`.

**Reply by kkoomen** [reply]
Thanks! I thought this was for any kind of GitHub feedback. Thanks.

**Reply by CherishSantoshi** [reply]
Hi, Was there any response from GitHub trending team on this question?
Would love to know more.

**Reply by marcelfolaron** [reply]
Following up. I would love to know more about how the Trending algorithm works. There are repositories that received a couple stars that show up on Trending while others receiving 30-50 Stars do not. It'd be good to understand what this algorithm is optimizing for.
